File: backend/models/job.js
module.exports = (sequelize, DataTypes) => {
  const Job = sequelize.define('Job', {
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    weight: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    dependent_on: {
      type: DataTypes.JSON,
      allowNull: true,
    },
    job_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    state: {
      type: DataTypes.STRING,
      allowNull: false,
      defaultValue: 'created', // Default value is 'created'
    }
  });

  return Job;
};

File: orchestrator/Models/Job.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Orchestrator.Models
{
    public class Job
    {
        [Key]
        public int Id { get; set; } // Primary key
        public int JobId { get; set; }
        public string Name { get; set; }
        public int Weight { get; set; }
        public List<int>? DependentOn { get; set; } = new List<int>();
        public string State { get; set; } = "NotStarted";

        // Foreign key property
        public int PlaybookId { get; set; }

        // Navigation property for the related playbook
        public Playbook Playbook { get; set; }
    }
}
